src/libs/ReportActionsUtils.ts

```diff
@@ -95,9 +95,12 @@ function isCreatedAction(reportAction: OnyxEntry<ReportAction>): boolean {
 }
 
 function isDeletedAction(reportAction: OnyxEntry<ReportAction | OptimisticIOUReportAction>): boolean {
-    // A deleted comment has either an empty array or an object with html field with empty string as value
     const message = reportAction?.message ?? [];
-    return message.length === 0 || message[0]?.html === '';
+
+    // A legacy deleted comment has either an empty array or an object with html field with empty string as value
+    const isLegacyDeletedComment = reportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.ADDCOMMENT && (!message.length || !message[0]?.html);
```

Comment:
Double checking, there's no regressions we might cause by adding`reportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.ADDCOMMENT`? Is it only comments that we have delete support for?

Reply:
for example, money requests?

Reply:
@grgia Thanks for your reminder. As I understand, @arosiclair added this condition to filter request money report action from `isLegacyDeletedComment` check. But It seems we missed [this case](https://github.com/Expensify/App/pull/36747#issuecomment-1994941714). We only should ignore `isLegacyDeletedComment` check if It is the only request in report

Reply:
@arosiclair Should we update this?

Reply:
> Double checking, there's no regressions we might cause by addingreportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.ADDCOMMENT? Is it only comments that we have delete support for?

Only report comments use this pattern of clearing the `message` so it shouldn't cause any issues. There are many actions with empty messages by default ([see here](https://expensify.slack.com/archives/C03TQ48KC/p1708114308456789?thread_ts=1708030880.121819&cid=C03TQ48KC)), so just checking for an empty message on any type of action was not correct.

> @arosiclair Should we update this?

I'm not sure understand your concern. When you're testing deleting expenses on a report, are you noticing any bugs or regressions? If so, can you write repro steps with your expected and actual outcomes?

Reply:
@arosiclair Currently, I don't see any regression. But let's see the video below






https://github.com/Expensify/App/assets/141406735/5521a941-281d-4ab2-aef7-b8e9998e5e51



I have 2 money requests. After I delete the second request, isDeletedAction still returns false

Reply:
Okay I see what you mean thanks for clarifying. It does look like a bug for those actions to still display after being deleted. We'll need more backend changes to address deleted expense actions, but that can be done later. 

I'll remove the ADDCOMMENT check so these get hidden and re-add it later once that's fixed.